basiclti/tsugi-util/src/test/org/tsugi/http/HttpUtilTest.java

```diff
@@ -45,5 +45,13 @@ public void testLinkExtractRel() {
 		String differencesUrl = HttpUtil.extractLinkByRel(headerValues, "differences");
 		assertEquals("https://lms.example.com/sections/2923/memberships?since=1422554502", differencesUrl);
 
+		// The Canvas way of Multiple HTTP headers with out a space after the comma
+		// This confuses the java.util.net.HttpHeaders.allValues()
+		// So we compensate.
+		headerValues = new ArrayList<String>();
+		headerValues.add("<https://umich.instructure.com/api/lti/courses/624214/names_and_roles?page=1&per_page=50>; rel=\"current\",<https://umich.instructure.com/api/lti/courses/624214/names_and_roles?page=2&per_page=50>; rel=\"next\",<https://umich.instructure.com/api/lti/courses/624214/names_and_roles?page=1&per_page=50>; rel=\"first\",<https://umich.instructure.com/api/lti/courses/624214/names_and_roles?page=3&per_page=50>; rel=\"last\"");
```

Comment:
I see you're including this per_page in the test cases. Is this configurable in the code? I think it having it configurable would make it easier to test (setting this value lower) and it should be set to the max by default for performance. [The documentation says there is an unspecified limit](https://canvas.instructure.com/doc/api/file.pagination.html) on how how big this value can be set but generally it seems like most API's it's 200.

Reply:
+1

Reply:
These test cases are just using the exact strings including key value parameters.  These parameters are not interpreted at all in the unit test.   And yes - I have seen defaults of 200 and defaults of 50.  The key when building SakaiPlus is not to assume "infinity" - which the Sakai code does handle if the Link: headers are formatted in a way that can be parsed.